src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -344,8 +344,8 @@ async def __umount(self, pool, uuid):
         for dataset, name in reversed(self.__get_datasets(pool, uuid)):
             try:
                 await run('umount', '-f', dataset)
-            except subprocess.CalledProcessError:
-                self.logger.warning('Unable to umount %r', dataset)
+            except subprocess.CalledProcessError as e:
+                raise CallError(f'Unable to umount {dataset}: {e.stderr.decode()}')
```

Comment:
I remember we were not able to unmount something sometimes but it was fine to ignore it, did it happen to you while working on this issue?

Reply:
It was not fine, we were just lazy to handle it correctly. In Linux -f does not really force anything.